python/ray/data/_internal/stats.py

```diff
@@ -407,6 +463,15 @@ def clear_iteration_metrics(self, dataset_tag: str):
         self.iter_total_blocked_s.set(0, tags)
         self.iter_user_s.set(0, tags)
         self.iter_initialize_s.set(0, tags)
+        self.iter_total_s.set(0, tags)
+        self.iter_wait_s.set(0, tags)
+        self.iter_get_s.set(0, tags)
+        self.iter_next_batch_s.set(0, tags)
+        self.iter_format_batch_s.set(0, tags)
+        self.iter_collate_batch_s.set(0, tags)
+        self.iter_finalize_batch_s.set(0, tags)
+        self.streaming_split_coordinator_s.set(0, tags)
```

Comment:
for the iteration related metrics, let's move them to `clear_iteration_metrics()`.